pkg/driver/node_server.go

```diff
@@ -448,11 +452,15 @@ func (driver *Driver) setupDevice(publishContext map[string]string) (*model.Devi
 		ConnectionMode: defaultConnectionMode,
 	}
 	if publishContext[accessProtocolKey] == iscsi {
-		chapInfo := &model.ChapInfo{
-			Name:     publishContext[chapUsernameKey],
-			Password: publishContext[chapPasswordKey],
+		// ChapPassword may not be returned from publishContext
+		// Validate the username from configMap and compare it from that returned from publishContext, and use the password from configMap
+		if publishContext[chapUsernameKey] == ChapUsername {
+			chapInfo := &model.ChapInfo{
+				Name:     ChapUsername,
+				Password: ChapPassword,
+			}
+			volume.Chap = chapInfo
```

Comment:
Why does this need to be in the publish context?  I don't think we need to store these credentials on a k8s object do we?  We pull them at ControllerPublishVolume time to give to the CSP for configuration and again at NodeStageVolume time when the device actually needs to be created.

Reply:
yes, this would be perfect since we have these values with both controller/node plugin already.

Reply:
i am changing this in CSP to return the credentials from host object so that there is no change in the csi driver (controller/node)